src/test/java/com/google/cloud/tools/appengine/cloudsdk/serialization/CloudSdkComponentTest.java

```diff
@@ -0,0 +1,154 @@
+/*
+ * Copyright 2016 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.tools.appengine.cloudsdk.serialization;
+
+import com.google.gson.Gson;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.runners.MockitoJUnitRunner;
+
+import java.util.Arrays;
+import java.util.List;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+
+/**
+ * Unit tests for {@link CloudSdkComponent}.
+ */
+@RunWith(MockitoJUnitRunner.class)
+public class CloudSdkComponentTest {
+
+  private Gson gson;
+
+  @Before
+  public void setup() {
+    this.gson = new Gson();
+  }
+
+  @Test
+  public void testGsonDeserializeJson() {
```

Comment:
It's testing that the class can be deserialized from json and that all of its expected fields are present, etc. The `CloudSdkComponent` class itself contains some Gson-specific annotations that make it deserialize properly when using the gson library, which is why I thought to include "Gson" in the test names. 

I can just rename to `testDeserialization`, etc if you prefer.